shared/src/commonMain/kotlin/org/hyperskill/app/core/injection/StateRepositoriesComponentImpl.kt

```diff
@@ -1,17 +1,30 @@
 package org.hyperskill.app.core.injection
 
+import org.hyperskill.app.study_plan.data.repository.CurrentStudyPlanStateRepositoryImpl
+import org.hyperskill.app.study_plan.domain.repository.CurrentStudyPlanStateRepository
+import org.hyperskill.app.study_plan.remote.StudyPlanRemoteDataSourceImpl
 import org.hyperskill.app.subscriptions.data.repository.CurrentSubscriptionStateRepositoryImpl
 import org.hyperskill.app.subscriptions.data.source.SubscriptionsRemoteDataSource
 import org.hyperskill.app.subscriptions.domain.repository.CurrentSubscriptionStateRepository
 import org.hyperskill.app.subscriptions.remote.SubscriptionsRemoteDataSourceImpl
 
 class StateRepositoriesComponentImpl(appGraph: AppGraph) : StateRepositoriesComponent {
+
+    private val authorizedHttpClient = appGraph.networkComponent.authorizedHttpClient
+
     /**
      * Current subscription
      */
     private val subscriptionsRemoteDataSource: SubscriptionsRemoteDataSource =
-        SubscriptionsRemoteDataSourceImpl(appGraph.networkComponent.authorizedHttpClient)
+        SubscriptionsRemoteDataSourceImpl(authorizedHttpClient)
 
     override val currentSubscriptionStateRepository: CurrentSubscriptionStateRepository =
         CurrentSubscriptionStateRepositoryImpl(subscriptionsRemoteDataSource)
+
+    /**
+     * Study plan
+     */
+    override val studyPlanStateRepository: CurrentStudyPlanStateRepository by lazy {
```

Comment:
@ivan-magda 
Because we don't need it until someone requests it.
On the first call repository will be created, and other requestors will get cached value.

Reply:
@ivan-magda 
Di graph is initialized just after the app startup. We need studyPlanStateRepository quite later.